### PR TITLE
[core] Support aggregation system table

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
### Purpose

Support Aggregation system table.

### CODE REVIEW

**Comment by MonsterChenzhuo** [comment]
@JingsongLi  PTAL,thanks.

File: docs/content/how-to/system-tables.md [hunk 1]
```diff
@@ -256,6 +256,25 @@ SELECT * FROM MyTable$manifests /*+ OPTIONS('scan.snapshot-id'='1') */;
...
+### Aggregation Table
+
+You can query the historical aggregation of the table through aggregation table.
+
+```sql
+SELECT * FROM MyTable$aggregation;
+
+/*
++------------+-----------------+---------------------+---------+
+| field_name |            type | AggregationFunction | comment |
```
Comment (JingsongLi) [inline docs/content/how-to/system-tables.md:268]: field_name
field_type
function
function_options
comment

What is function_options? For example, fields.xx.min or fields.xx.max.

File: paimon-core/src/main/java/org/apache/paimon/table/system/AggregationTable.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+import java.util.Objects;
+
+import static org.apache.paimon.catalog.Catalog.SYSTEM_TABLE_SPLITTER;
+
+/** A {@link Table} for showing Aggregation of table. */
+public class AggregationTable implements ReadonlyTable {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final String AGGREGATION = "aggregation";
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/table/system/AggregationTable.java:62]: aggregation_fields?

File: docs/content/how-to/system-tables.md [hunk 3]
```diff
@@ -256,6 +256,25 @@ SELECT * FROM MyTable$manifests /*+ OPTIONS('scan.snapshot-id'='1') */;
...
+### Aggregation fields Table
+
+You can query the historical aggregation of the table through aggregation fields table.
+
+```sql
+SELECT * FROM MyTable$aggregation_fields;
+
+/*
++------------+-----------------+-------------+-----------------------+---------+
+| field_name |      field_type |    function | function_options      | comment |
```
Comment (JingsongLi) [inline docs/content/how-to/system-tables.md:268]: test `function_options` has two options.

Comment (MonsterChenzhuo) [reply docs/content/how-to/system-tables.md]: done.

File: docs/content/how-to/system-tables.md (hunk 1 above)
Comment (MonsterChenzhuo) [reply docs/content/how-to/system-tables.md]: done.

File: paimon-core/src/main/java/org/apache/paimon/table/system/AggregationTable.java (hunk 2 above)
Comment (MonsterChenzhuo) [reply paimon-core/src/main/java/org/apache/paimon/table/system/AggregationTable.java]: done.

**Comment by MonsterChenzhuo** [comment]
@JingsongLi PTAL,thanks.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**